plugin/trino-hive/src/main/java/io/trino/plugin/hive/orc/OrcFileWriter.java

```diff
@@ -211,6 +212,18 @@ public void commit()
                 throw new TrinoException(HIVE_WRITE_VALIDATION_FAILED, e);
             }
         }
+
+        Callable<Void> rollbackAction = this.rollbackAction;
+        return new RollbackAction(
+                () -> {
+                    try {
+                        rollbackAction.call();
+                    }
+                    catch (Exception e) {
+                        throw new TrinoException(HIVE_WRITER_CLOSE_ERROR, "Error rolling back write to Hive", e);
+                    }
+                },
+                this);
```

Comment:
By giving `RollbackAction` a reference to this object aren't we still retaining this object but now without accounting for it ?